src/main/scala/devsearch/features/FeatureRecognizer.scala

```diff
@@ -87,7 +87,13 @@ object FeatureRecognizer extends (CodeFile => TraversableOnce[Feature]) with jav
     ValDefExtractor
   )
 
-  def apply(data: CodeFile) = extractors.flatMap(_.extract(data))
+  def apply(data: CodeFile): Set[Feature] = {
+    extractors.flatMap { case extractor =>
+      scala.util.Try {
```

Comment:
This will simply catch any type of exception, right? Wouldn't it be better to be more specific so we don't get hidden crashes?